overlord/patch/patch_test.go

```diff
@@ -456,3 +457,42 @@ func generatePatchFunc(testValue int, sequence *[]int) patch.PatchFunc {
 		return nil
 	}
 }
+
+func (s *patchSuite) TestRegressionCoreCurrentSymlinkMissing(c *C) {
+	// simulate missing "core" symlink
+	dirs.SetRootDir(c.MkDir())
+	defer func() { dirs.SetRootDir("/") }()
+
+	log, restore := logger.MockLogger()
+	defer restore()
+
+	st := state.New(nil)
+	patch.Init(st)
+
+	st.Lock()
+	siCore1 := &snap.SideInfo{RealName: "core", Revision: snap.R(1)}
+	siCore2 := &snap.SideInfo{RealName: "core", Revision: snap.R(2)}
+	snapstate.Set(st, "core", &snapstate.SnapState{
+		SnapType: "os",
+		Active:   true,
+		Sequence: []*snap.SideInfo{siCore1, siCore2},
+		Current:  siCore1.Revision,
+	})
+	st.Unlock()
+
+	subLevel := 1
+	err := patch.MaybeResetSublevelForLevel60(st, &subLevel)
+	c.Assert(err, IsNil)
+	c.Assert(log.String(), Matches, `(?m).*WARNING: cannot determine core refresh time: lstat /.*/snap/core/current: no such file or directory`)
+	c.Assert(subLevel, Equals, 0)
+
+	st.Lock()
+	defer st.Unlock()
+	var stateSublevel int
+	var stateSublevelReset time.Time
+	c.Assert(st.Get("patch-sublevel", &stateSublevel), IsNil)
+	c.Assert(st.Get("patch-sublevel-reset", &stateSublevelReset), IsNil)
+	c.Check(stateSublevel, Equals, 0)
+	// sublevelReset time is not updated
+	c.Check(stateSublevelReset.IsZero(), Equals, true)
```

Comment:
It actually *is* updated (with lastRefresh=0), just  happens to be zero before in the test. I'd modify the test slightly to better cover this condition, i.e. set "patch-sublevel-reset" to a non-zero time, and afterwards check if it's zero.

Reply:
Indeed, that makes perfect sense